k8s.go

```diff
@@ -0,0 +1,247 @@
+}
+
+// KubernetesCluster kubernetes cluster
+type KubernetesCluster struct {
+	// URL to the object representation (absolute path)
+	// Read Only: true
+	// Format: uri
+	Href string `json:"href,omitempty"`
+
+	// The resource's unique identifier.
+	// Read Only: true
+	ID string `json:"id,omitempty"`
+
+	// metadata
+	Metadata *NoStateMetaData `json:"metadata,omitempty"`
+
+	// properties
+	// Required: true
+	Properties *KubernetesClusterProperties `json:"properties"`
+
+	// The type of object
+	// Read Only: true
+	// Enum: [k8s]
+	PBType string `json:"type,omitempty"`
+
+	// Entities of a cluster
+	Entities KubernetesClusterEntities `json:"entities"`
+}
+
+type KubernetesClusterEntities struct {
+	// NodePools of a cluster
+	NodePools *KubernetesNodePools `json:"nodepools,omitempty"`
+}
+
+// KubernetesClusterProperties kubernetes cluster properties
+type KubernetesClusterProperties struct {
+	// A Kubernetes Cluster Name. Valid Kubernetes Cluster name must be 63 characters or less and must be empty
+	// or begin and end with an alphanumeric character ([a-z0-9A-Z]) with dashes (-), underscores (_), dots (.),
+	// and alphanumerics between.
+	// Required: true
+	Name string `json:"name"`
+}
+
+// KubernetesConfig kubernetes config
+type KubernetesConfig struct {
+	// URL to the object representation (absolute path)
+	// Read Only: true
+	// Format: uri
+	Href string `json:"href,omitempty"`
+
+	// The resource's unique identifier.
+	// Read Only: true
+	ID string `json:"id,omitempty"`
+
+	// properties
+	// Required: true
+	Properties *KubernetesConfigProperties `json:"properties"`
+
+	// The type of object
+	// Read Only: true
+	// Enum: [kubeconfig]
+	PBType string `json:"type,omitempty"`
+}
+
+type KubernetesConfigProperties struct {
+	// A Kubernetes Config file data
+	KubeConfig string `json:"kubeconfig,omitempty"`
+}
+
+type KubernetesNodePool struct {
+	// URL to the object representation (absolute path)
+	// Read Only: true
+	// Format: uri
+	Href string `json:"href,omitempty"`
+
+	// The resource's unique identifier.
+	// Read Only: true
+	ID string `json:"id,omitempty"`
+
+	// metadata
+	Metadata *Metadata `json:"metadata,omitempty"`
+
+	// properties
+	// Required: true
+	Properties *KubernetesNodePoolProperties `json:"properties"`
+
+	// The type of object
+	// Read Only: true
+	// Enum: [nodepool]
+	PBType string `json:"type,omitempty"`
+}
+
+// KubernetesNodePoolProperties kubernetes node pool properties
+// swagger:model KubernetesNodePoolProperties
+type KubernetesNodePoolProperties struct {
+	// The availability zone in which the server should exist
+	// Required: true
+	// Enum: [AUTO ZONE_1 ZONE_2]
+	AvailabilityZone string `json:"availabilityZone,omitempty"`
+
+	// Number of cores for node
+	// Required: true
+	CoresCount int64 `json:"coresCount,omitempty"`
+
+	// A valid cpu family name
+	// Required: true
+	CPUFamily string `json:"cpuFamily,omitempty"`
+
+	// A valid uuid of the datacenter on which user has access
```

Comment:
Yeah, but mainly where the nodepool is provisioned. The official description "The unique identifier of the data center where the worker nodes of the node pool are provisioned." fits quite well.